opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java

```diff
@@ -2,39 +2,48 @@
 
 import androidx.annotation.NonNull;
 
+import net.sqlcipher.Cursor;
+
 import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.Contract;
 import org.smartregister.chw.domain.EligibleChild;
 import org.smartregister.chw.domain.VillageDose;
-import org.smartregister.chw.model.FilterReportFragmentModel;
 import org.smartregister.chw.util.ReportingConstants;
 import org.smartregister.dao.AbstractDao;
 
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
-import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.TreeMap;
 
+import timber.log.Timber;
+
 /**
  * @author rkodev
  */
 public class ReportDao extends AbstractDao {
 
     @NonNull
-    public static List<String> extractRecordedLocations() {
-        String sql = "select distinct location_id from ec_family_member_location";
-
-        AbstractDao.DataMap<String> dataMap = c -> getCursorValue(c, "location_id");
-        List<String> res = AbstractDao.readData(sql, dataMap);
-        if (res == null || res.size() == 0)
-            return new ArrayList<>();
-
-        return res;
+    public static HashMap<String, String> extractRecordedLocations() {
+        HashMap<String, String> hashMap = new HashMap<>();
+        try {
+            String query = "SELECT DISTINCT location_id, provider_id FROM ec_family_member_location";
```

Comment:
You can use a DataMap object to read the cursor. If not male sure to close the cursor objectt